plugin/src/main/kotlin/me/tylerbwong/gradle/metalava/Module.kt

```diff
@@ -10,54 +9,90 @@ import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
 import java.io.File
 
 internal sealed class Module {
-
+    /**
+     * The bootclasspath to be passed to metalava when it parses the source files.
+     *
+     * @return The bootclasspath for to be passed to metalava. May be empty.
+     *
+     * @see compileClasspath
+     */
     open val bootClasspath: Collection<File> = emptyList()
-    abstract val compileClasspath: Collection<File>
 
-    class Android(private val extension: LibraryExtension, private val variantName: String) : Module() {
+    /**
+     * Returns directories and/or files containing classes that are to be on the classpath when
+     * metalava parses the source files.
+     *
+     * Depending on the Platform, different classpaths may be required, and these can be identified
+     * with the [variant] parameter.
+     *
+     * @param variant Identifies a classpath retained by the Module. If `null` then the default
+     * classpath for that Platform is used.
+     * @return The classpath for to be passed to metalava. May be empty.
+     *
+     * @see variants
+     */
+    abstract fun compileClasspath(variant: String? = null): Collection<File>
+
+    /**
+     * The list of available variants to be passed into [compileClasspath] so as to filter
+     * its output. Can be null.
+     *
+     * @see compileClasspath
+     */
+    open val variants: Collection<String>? = null
```

Comment:
I am still not sure how I feel about introducing `variants` here as it is only specific to Android.